src/config_store.rs

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate serde_json;
+
+use std::collections::BTreeMap;
+use std::fs;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::sync::Mutex;
+
+
+type ConfigNameSpace = BTreeMap<String, String>;
+
+type ConfigTree = BTreeMap<String, ConfigNameSpace>;
+
+#[derive(Debug)]
+pub struct ConfigStore {
+    file_name: String,
+    save_lock: Mutex<()>,
+    config: ConfigTree
+}
+
+impl ConfigStore {
+    pub fn new(file_name: String) -> Self {
+        ConfigStore {
+            file_name: file_name.clone(),
+            save_lock: Mutex::new(()),
+            config: ConfigStore::load(file_name)
+        }
+    }
+
+    pub fn set(&mut self, namespace: String, property: String, value: String) {
```

Comment:
Could all these parameters be &str instead? that may help with not cloning at call sites